xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h

```diff
@@ -2017,24 +2017,51 @@ class ATTR_DLL_LOCAL CFile
   /// @param[out] line The buffer to store the data in.
   /// @return True when a line was read, false otherwise.
   ///
+  ///
+  /// ------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/Filesystem.h>
+  ///
+  /// ...
+  ///
+  /// /* Create the needed file handle class */
+  /// kodi::vfs::CFile myFile;
+  ///
+  /// /* Open the wanted file */
+  /// if (myFile.OpenFile(kodi::addon::GetUserPath("/myFile.txt")))
+  /// {
+  ///   /* Read all lines inside file */
+  ///   while (1)
+  ///   {
+  ///     std::string line;
+  ///     if (!myFile.ReadLine(line))
+  ///       break;
+  ///     fprintf(stderr, "%s\n", line.c_str());
+  ///   }
+  /// }
+  ///
+  /// ~~~~~~~~~~~~~
+  ///
   bool ReadLine(std::string& line)
   {
     using namespace kodi::addon;
 
     line.clear();
     if (!m_file)
       return false;
-    // TODO: Read 1024 chars into buffer. If file position advanced that many
-    // chars, we didn't hit a newline. Otherwise, if file position is 1 or 2
-    // past the number of bytes read, we read (and skipped) a newline sequence.
-    char buffer[1025];
-    if (CPrivateBase::m_interface->toKodi->kodi_filesystem->read_file_string(
-            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer)))
+
+    const auto toKodi = CPrivateBase::m_interface->toKodi;
+
+    char buffer[1024];
+    buffer[sizeof(buffer) - 1] = 0; // Force to have always 0 at end
+    while (toKodi->kodi_filesystem->read_file_string(toKodi->kodiBase, m_file, buffer,
+                                                     sizeof(buffer) - 2))
```

Comment:
I still think the minus 2 should be minus 1. This is saying read at most 1022 bytes into a 1024 byte buffer, meaning that char index `1022` will always be junk (`1023` is `0` and `1021` is `0` because the function reads a C-string)

Reply:
I can ask the user who is suffering from this issue to test it with the -1 instead, I’d imagine it will still work.

Reply:
I changed this to -1 and users are testing it now.